Added section on wavefunction ordering to QVM docs

ntezak opened the pull request:
This fixes #240

ntezak commented:
These are the rendered docs:
![image](https://user-images.githubusercontent.com/1573961/34169643-ece68a5a-e49c-11e7-941d-4d47f3100a53.png)

ntezak commented:
@stevenheidel I was not sure if this was the best place for the docs. We could add a more explicit diagram visualizing the qubit ordering and also add a link to this from the ``Program.defgate`` API call.

stevenheidel commented:
I think that's a good spot for it, the qvm overview page didn't have a lot of content on it yet.

Yes I like your idea, let's link to it from the defgate constructor and perhaps from the .gate call too if you think that makes sense.

ntezak commented:
Fleshed out docs a bit more and added cross references from various places (Wavefunction, Program.gate, Program.defgate, Program.def_noisy_gate).
Also added a commit that fixed some sphinx warnings in nearby code (sorry for sneaking that in)
